src/clientrpc.cpp

```diff
@@ -94,11 +98,127 @@ CClientRpc::CClientRpc ( CClient* pClient, CRpcServer* pRpcServer, QObject* pare
                                             } );
     } );
 
+    /// @rpc_notification jamulusclient/serverListReceived
+    /// @brief Emitted when the server list is received.
+    /// @param {array} params.servers - The server list.
+    /// @param {string} params.servers[*].address - Socket address (ip_address:port)
+    /// @param {string} params.servers[*].name - Server name
+    /// @param {number} params.servers[*].countryId - Server country ID (see QLocale::Country).
+    /// @param {string} params.servers[*].country - Server country
+    /// @param {string} params.servers[*].city - Server city
+    connect ( pClient->getConnLessProtocol(),
+              &CProtocol::CLServerListReceived,
```

Comment:
How often is an RPC client going to receive this message?

Given the nature of the server list (i.e. it's not guaranteed to be in any particular order or even complete), in what use cases would this message be consumed?  (Just thinking the JSON RPC doc might be expanded to explain - it's a bit brief currently.)

Reply:
The list seems to be sent only after being requested by the poll. I am using it to populate a list as required on the basis that the list will be accurate at that time. It seems to work well.

Reply:
It also gets issued if the connect dialog is open.